### DISCUSSION TITLE
Signup with google

### DISCUSSION THREAD

**Discussion by ricardosikic** [supabase/supabase#14770] [Questions] [answered]
Hi, 
is there a way to signup with google and get back the user information to update other table. Like supabase actually do with this function.

```
 const { user, error, session } = await supabase.auth.signUp({
            email: email,
           password: password,
      },
      {
        data: { 
          username: username
        }
      }
    );
```
For sign up with google i did this function that allow the user create an account but, i cannot get back the user information 

```
 const signUpWithGoogle = async() => {
    if(isPrivacyAccepted === true) {
      try {
        const { data, user, session, error } = await supabase.auth.signIn({
          provider: 'google',
        },
        {
          redirectTo: process.env.NODE_ENV === 'development' ? 'http://localhost:3000/posts' : 'https://lalalalal.app/la'
        });
        if(error) throw error;
      
        if(user) {
          addAcceptedPrivacyPolicy(user);
        }
      } catch (error) {
        notifyError(error);
      }  
    } else {
      setPrivacyAcceptedMessage(true);
    }
  }
```

**Comment by GaryAustin1** [comment, marked as answer]
No.  The OAuth process only returns the URL it is going to be redirected to.  You can't get a user on the return because the user is not known yet, and the browser is about to be redirected to the Google login screen.  When the Google part is done you get redirected back to the redirectTo URL.

**Reply by ricardosikic** [reply]
thanks for the answer.
